### PR TITLE
Auth frontend state

### PR DESCRIPTION
**Pull request by soapraj**
- This is how the auth flow works. Only on page refresh we can pick up the profile information of the user.

auth-state-flow

- This implementation only considers Google Auth for the auth state in the sidebar. 

- The app config would have to list all the auth providers and an identity provider so that we can list all the providers and show profile information only when the user is authenticated with the identity provider.

- In case of google it is both access and identity provider but it other cases there could be separate identity and access providers.

Raising this draft PR to gather some feedback and thoughts.

#### :heavy_check_mark: Checklist

- [ ] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

File: plugins/auth-backend/src/providers/PassportStrategyHelper.ts [hunk 1]
```diff
@@ -98,10 +134,22 @@ export const executeRefreshTokenStrategy = async (
             ),
           );
         }
-        resolve({
+
+        anyStrategy.userProfile(
```
Comment (Rugvip) [inline plugins/auth-backend/src/providers/PassportStrategyHelper.ts:138]: Talked about exposing this as a separate helper to let it be a bit more flexible in the provider implementation

File: plugins/auth-backend/src/providers/PassportStrategyHelper.ts [hunk 2]
```diff
@@ -16,7 +16,43 @@
 
 import express from 'express';
 import passport from 'passport';
-import { RedirectInfo, RefreshTokenResponse } from './types';
+import jwtDecoder from 'jwt-decode';
+import { RedirectInfo, RefreshTokenResponse, ProfileInfo } from './types';
+
+export const makeProfileInfo = (
```
Comment (Rugvip) [inline plugins/auth-backend/src/providers/PassportStrategyHelper.ts:22]: Bit worried that this is gonna be a bit too different between providers, but let's hit that problem and deal with it later :grin:

File: packages/storybook/.storybook/apis.js [hunk 3]
```diff
@@ -13,4 +17,18 @@ const alertApi = builder.add(alertApiRef, new AlertApiForwarder());
 
 builder.add(errorApiRef, new ErrorAlerter(alertApi, new ErrorApiForwarder()));
 
+const oauthRequestApi = builder.add(
```
Comment (Rugvip) [inline packages/storybook/.storybook/apis.js:20]: 🎉

File: packages/core-api/src/apis/definitions/auth.ts [hunk 4]
```diff
@@ -143,6 +143,30 @@ export type OpenIdConnectApi = {
...
+   */
+  optional?: boolean;
+};
+
+export type ProfileInfoApi = {
+  getProfile(options?: ProfileInfoOptions): Promise<ProfileInfo | undefined>;
+};
+
+export type ProfileInfo = {
+  provider: string;
```
Comment (Rugvip) [inline packages/core-api/src/apis/definitions/auth.ts:164]: I'd say we can remove provider from the info, I don't think you're using it either? With the current API you'll always know which provider you're dealing with when it's being consumed, and if you want to treat the in a more generic way at some point you could add in the information later, e.g.

```ts
const allTheProviders: {api: ProfileInfoApi, provider: string}[] = [
  {api: googleAuthApi, provider: 'google'},
  {api: githubAuthApi, provider: 'github'},
]
```

File: packages/core-api/src/apis/implementations/auth/google/GoogleAuth.ts [hunk 5]
```diff
@@ -39,6 +41,7 @@ type CreateOptions = {
 };
 
 export type GoogleAuthResponse = {
+  profile: any;
```
Comment (Rugvip) [inline packages/core-api/src/apis/implementations/auth/google/GoogleAuth.ts:44]: I think you can use the same profile type here too right?

**Review by Rugvip — COMMENTED** [review]
Looks good 👍 , couple of things to move around, as discussed

**Comment by stefanalund** [comment]
Fixes #1044

**Review by marcuseide — APPROVED** [review]
Discussing this with @nikek and we feel it would be easier to merge this as-is and continue working on it.

**Comment by stefanalund** [comment]
🚢

**Pull request merged**